### PR TITLE
chore(deps): bumping cdk from 1.75.0 to 1.78.0

### PR DESCRIPTION
**Pull request by horsmand**
I deployed the all in AWS infrastructure basic app to confirm that everything is working.

Besides the CDK version bump, this includes:
* Bumping the Deadline version used in the AWS infrastructure basic app example (it was on 10.1.11.5 which was pulled)
* Some new linting rules for our tests based on a recent change that CDK made: https://github.com/aws/aws-cdk/commit/faee6a83fdb1d17d79d93e5d63bd7a2d1893c571#diff-9e9a2927b07e99f25479b6e61eee21aaad25d2a4721c27a74d2b7159b21b0c8d
* The fixes required to upgrade `typescript` that were failing this PR: https://github.com/aws/aws-rfdk/pull/244
* The fixes required to upgrade `@types/yargs` that were failing this PR: https://github.com/aws/aws-rfdk/pull/256
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by ddneilson — DISMISSED** [review]

**Review by ddneilson — DISMISSED** [review]

File: examples/deadline/All-In-AWS-Infrastructure-Basic/ts/bin/config.ts [hunk 1]
```diff
@@ -15,11 +15,11 @@ import { MongoDbSsplLicenseAcceptance } from 'aws-rfdk';
 class AppConfig {
 
   /**
-   * A map of regions to Deadline Client Linux AMIs. As an example, the Linux Deadline 10.1.11.5 AMI ID from us-west-2
+   * A map of regions to Deadline Client Linux AMIs. As an example, the Linux Deadline 10.1.12.1 AMI ID from us-west-2
```
Comment (ryyakobe) [inline examples/deadline/All-In-AWS-Infrastructure-Basic/ts/bin/config.ts:18]: Maybe we should also remove deadline 10.1.11.5 from `aws-rfdk/integ/scripts/bash/rfdk-integ-e2e.sh.`

File: examples/deadline/All-In-AWS-Infrastructure-Basic/ts/package.json [hunk 2]
```diff
@@ -18,13 +18,13 @@
...
   },
   "devDependencies": {
-    "@types/node": "^14.14.10",
-    "aws-cdk": "1.75.0",
-    "ts-node": "^9.0.0",
-    "typescript": "~4.0.5"
+    "@types/node": "^14.14.14",
+    "aws-cdk": "1.78.0",
+    "ts-node": "^9.1.1",
+    "typescript": "~4.1.3"
```
Comment (ryyakobe) [inline examples/deadline/All-In-AWS-Infrastructure-Basic/ts/package.json:24]: I wonder why aws-cdk is not bumping their typescript version from 3.9.7. But as long as it works for us, we should be fine upgrading it.

File: integ/package.json [hunk 3]
```diff
@@ -71,38 +71,38 @@
...
+    "@aws-cdk/aws-elasticloadbalancingv2": "1.78.0",
+    "@aws-cdk/aws-iam": "1.78.0",
+    "@aws-cdk/aws-logs": "1.78.0",
+    "@aws-cdk/aws-route53": "1.78.0",
+    "@aws-cdk/aws-s3": "1.78.0",
+    "@aws-cdk/aws-s3-assets": "1.78.0",
+    "@aws-cdk/aws-secretsmanager": "1.78.0",
+    "@aws-cdk/core": "1.78.0",
     "aws-rfdk": "0.21.0",
-    "aws-sdk": "^2.799.0"
```
Comment (ryyakobe) [inline integ/package.json:105]: We didn't update versions for `aws-sdk`, `typescript`, and  `@types/node` in `aws-rfdk/lambda-layers/package.json`. Do we need to update them?

**Review by jusiskin — DISMISSED** [review]

File: examples/deadline/All-In-AWS-Infrastructure-Basic/ts/package.json (hunk 2 above)
Comment (horsmand) [reply examples/deadline/All-In-AWS-Infrastructure-Basic/ts/package.json:24]: Their codebase is so large I bet they have a lot more breaking changes to work through than we did.

File: integ/package.json (hunk 3 above)
Comment (horsmand) [reply integ/package.json:105]: Ah good catch, I missed this directory. I'll update this.

File: examples/deadline/All-In-AWS-Infrastructure-Basic/ts/bin/config.ts (hunk 1 above)
Comment (horsmand) [reply examples/deadline/All-In-AWS-Infrastructure-Basic/ts/bin/config.ts:18]: Updated

**Review by jusiskin — APPROVED** [review]

**Review by ryyakobe — APPROVED** [review]
Thanks David!

**Pull request merged**